lib/active_record/connection_adapters/sqlserver/database_statements.rb

```diff
@@ -104,6 +138,21 @@ def execute_procedure(proc_name, *variables)
           name = 'Execute Procedure'
           log(sql, name) do
             case @connection_options[:mode]
+            when :jdbc
+              result = begin
+                @connection.fetch(sql).all
+              rescue => e
```

Comment:
This rescue feels risky. Why would an empty result set to fetch raise an exception? Is it due to the `.all` method call? If so, can we avoid the rescue trap?

Reply:
I have a similar concern to @metaskills. Is there a reason we're rescuing all exceptions as "statement invalid", instead of just the ones the method throws that we know to mean "statement invalid"?